pylint/checkers/spelling.py

```diff
@@ -368,6 +384,7 @@ def _check_spelling(self, msgid, line, line_num):
             starts_with_comment = False
 
         line = _strip_code_flanked_in_backticks(line)
```

Comment:
@eli88fine Sorry should have been clearer. I was referring to this function!

Reply:
ah, you're saying see if I can somehow combine the regular expressions into a single one.  I understand now.  hmm, I'll take a crack at it and see how it goes. that would definitely be the most complex thing I've ever tried with regex

Reply:
hmm, I'm worried about accidentally creating a bunch of weird edge cases by trying to combine them.

What if I had a "pre-filter" that just checked the line for either backticks OR a `type: ignore`.  Then if that hit it would run both of the existing regex statements

`if '`' in line or 'type: ignore' in line`:`

Reply:
Perhaps you are right. Let's leave as is. I'm not sure if the performance benefits of skipping `regex` matching outweigh the downsides of extra `ifs`.